### DISCUSSION TITLE
How do you access `faker` outside of a factory?

### DISCUSSION THREAD

**Discussion by SpencerKaiser** [mikro-orm/mikro-orm#3195] [Q&A] [answered]
I have a `Seeder` that randomly creates related entities and occasionally I need to use `Faker` inside of my `Seeder` instead of in my `Factory`. Is there a way to do that?

Creating a new instance of `Faker` with `new Faker()` seems to break the factories and makes the `faker` they receive `undefined`

**Comment by B4nan** [comment, marked as answer]
> Creating a new instance of Faker with new Faker() seems to break the factories and makes the faker they receive undefined

That sounds more than weird, I kinda doubt instancing it again would break things. We don't even instance it ourselves, we use the default instance, and we reexport it too.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/seeder/src/index.ts#L13

**Reply by SpencerKaiser** [reply]
I mean I thought it was stupid weird too, but it was reproducible on my end... just drop a `const faker = new Faker()` into the top of your seeder file and then use the faker provided to your factory and 💥 

I knew that wasn't the right approach though, just didn't know how to do the above. Want me to update the docs to include the solution above?

Also do you have recommendations for how to link things together properly? We just batch stuff up in our seeder file and then link things as needed within that file. It works, but I question whether there's a better way to do it.

Also as usual, your stuff is flawless 👌 the difficulty of seeding and maintaining seeder files is trivial... awesome job!

**Reply by B4nan** [reply]
Docs improvements are always welcome :] Also worth checking the latest dev version where we have newer faker version, maybe its something fixed (I doubt its something to fix on our end, we really just pass the default instance).

For building relations between generated entities you could use the [shared context argument](https://mikro-orm.io/docs/seeding#shared-context)
